--- CMakeLists.txt	2019-02-06 20:13:51.000000000 +0100
+++ /Users/danieleferla/Downloads/llvm-7.0.0.src/lib/Target/X86/CMakeLists.txt	2018-07-13 13:13:58.000000000 +0200
@@ -65,12 +65,6 @@
   X86VZeroUpper.cpp
   X86WinAllocaExpander.cpp
   X86WinEHState.cpp
-  ropfuscator/X86ROPfuscator.cpp
-ropfuscator/BinAutopsy.cpp
-ropfuscator/XchgGraph.cpp
-ropfuscator/CapstoneLLVMAdpt.cpp
-ropfuscator/LivenessAnalysis.cpp
-ropfuscator/X86ROPUtils.cpp
   )
 
 add_llvm_target(X86CodeGen ${sources})
@@ -81,25 +75,3 @@
 add_subdirectory(MCTargetDesc)
 add_subdirectory(TargetInfo)
 add_subdirectory(Utils)
-
-# Adding capstone
-find_package(capstone QUIET)
-
-if(NOT capstone_FOUND)
-	find_package(PkgConfig REQUIRED)
-	if(PKG_CONFIG_FOUND)
-		pkg_search_module(capstone capstone REQUIRED)
-	endif()
-endif()
-
-if(NOT capstone_FOUND)
-	message(FATAL_ERROR "Unable to find capstone. Please install pkg-config and capstone development files")
-endif()
-
-# Linking da man
-target_link_libraries(${CURRENT_LLVM_TARGET} LINK_PRIVATE capstone)
-
-
-
-# Linking da man
-target_link_libraries(${CURRENT_LLVM_TARGET} LINK_PRIVATE bfd)
